index.js

```diff
@@ -1,2 +1,11 @@
 'use strict';
-module.exports = (cond, doIf, doElse) => val => cond ? doIf(val) : (doElse ? doElse(val) : val);
+
+const isFn = require('is-fn');
+
+module.exports = (cond, doIf, doElse) => val => {
+	if (isFn(cond)) {
```

Comment:
It should be a function that can also return a promise that is awaited. See how I solved that here: https://github.com/sindresorhus/p-catch-if/blob/454fa343bbf147db4b00c0ba6fb2e103e997e570/index.js#L18-L20

Feel free to steal the tests from here too: https://github.com/sindresorhus/p-catch-if/blob/454fa343bbf147db4b00c0ba6fb2e103e997e570/test.js#L68-L89 (Needs to be adjusted though)